login/js/validation.js

```diff
@@ -0,0 +1,129 @@
+    nicknameError.textContent = "닉네임을 입력해주세요.";
+    nickname.classList.add('error')
+  } else {
+    nicknameError.textContent = "";
+    nickname.classList.remove('error');
+  } 
+}
+
+const password = document.getElementById('password');
+const passwordError = document.getElementById('passwordError');
+
+function checkPassword() {
+  if (password.value.length < 8) {
+    passwordError.textContent = "비밀번호를 8자 이상 입력해주세요.";
+    password.classList.add('error')
+  } else {
+    passwordError.textContent = "";
+    password.classList.remove('error');
+  } 
+}
+
+const comparePassword = document.getElementById('comparePassword');
+const comparePasswordError = document.getElementById('comparePasswordError');
+
+function comparePasswords() {
+  if (password.value.length < 8) {
+    comparePasswordError.textContent = "비밀번호를 8자 이상 입력해주세요.";
+    comparePassword.classList.add('error')
+  } else if (password.value !== comparePassword.value ) {
+    comparePasswordError.textContent = "비밀번호가 일치하지 않습니다.";
+    comparePassword.classList.add('error')
+  } else {
+    comparePasswordError.textContent = "";
+    comparePassword.classList.remove('error');
+  } 
+}
+
+const loginButton = document.getElementById('loginButton');
+
+function activationLogin2() {
+  const emailValue = emailInput.value.trim(); 
+  const passwordValue = password.value.trim(); 
+  const nicknameValue = nickname.value.trim();
+  const comparePasswordValue = comparePassword.value.trim();
+  if (regex.test(emailValue) &&
+    passwordValue.length >= 8 &&
+    nicknameValue.length > 1 &&
+    passwordValue === comparePasswordValue 
+  ) { 
+    loginButton.disabled = false
+    loginButton.classList.add('active');
+  } else {
+    loginButton.disabled = true
+    loginButton.classList.remove('active');
+  }  
+}
+
+function activationLogin() {
+  const emailValue = emailInput.value.trim(); 
+  const passwordValue = password.value.trim(); 
+  if (regex.test(emailValue) &&
+    passwordValue.length >= 8
+  ) { 
+    loginButton.disabled = false
+    loginButton.classList.add('active');
+  } else {
+    loginButton.disabled = true
+    loginButton.classList.remove('active');
+  }  
+}
+
+const eyeButton = document.getElementById('eyeButton')
+const eyeButton2 = document.getElementById('eyeButton2')
+ 
+function checkingPassword(){
+  const currentType = password.getAttribute("type");
+  if (currentType === "password") {
+    password.setAttribute("type", "text");
+    eyeButton.src = "../img/open-eye.png"; 
+  } else {
+    password.setAttribute("type", "password");
+    eyeButton.src = "../img/close-eye.png"; 
+  }
+}
+
+function checkingPassword2(){
+  const currentType = comparePassword.getAttribute("type");
+  if (currentType === "password") {
+    comparePassword.setAttribute("type", "text");
+    eyeButton2.src = "../img/open-eye.png"; 
+  } else {
+    comparePassword.setAttribute("type", "password");
+    eyeButton2.src = "../img/close-eye.png"; 
+  }
+}
```

Comment:
- 두 함수의 다른 점은 비교할 요소(password, comparePassword)와 적용될 요소가 다르다(eyeButton, eyeButton2)는 부분이네요.
해당 요소 혹은 필요한 값을 함수의 인자로 받는다면 하나의 함수로 합쳐볼 수 있습니다.
- 함수명은 함수의 동작을 나타내는 동사형으로 지어주시는 게 좋습니다.
`e.g. togglePasswordView`

```suggestion
function togglePasswordView(currentType, eyeElement) {
  if (currentType === "password") {
    password.setAttribute("type", "text");
    eyeElement.src = "../img/open-eye.png"; 
  } else {
    password.setAttribute("type", "password");
    eyeElement.src = "../img/close-eye.png"; 
  }
}
```

이후 해당 함수를 이런 식으로 사용할 수 있습니다.
```js
togglePasswordView(password.getAttribute("type"), eyeButton)
```

Reply:
그리고 이와 같이 공통적으로 사용할 수 있는 유틸 함수가 아니라면, validation.js에 한데 모아서 사용하는 대신 login.js와 signup.js 각각에서 선언하고 사용하는 게 오히려 유지보수에 좋을 것 같다는.. 개인적인 의견을 드립니다.
그 이유는 다음과 같습니다.
1. 현재는 각 페이지에 적용되는 로직을 여러 js 파일을 거쳐서 확인해야 한다(공통 로직이 아니더라도)
2. validation.js에서 공통 유틸 함수를 관리한다면 각 파일의 역할이 명확하다

물론 지금 작성하신 방법도 잘못된 건 아닙니다!